Dockerfile

```diff
@@ -115,30 +121,36 @@ lapack_libs = mkl_lapack95_lp64' >/root/.numpy-site.cfg && \
     rm -rf /var/lib/apt/lists/* && \
     rm -rf /usr/share/doc/*
 
-ADD server/requirements.txt /server/requirements.txt
+
+RUN useradd -U -ms /bin/bash worker
+
+ADD --chown=worker:worker server/requirements.txt /server/requirements.txt
+
 ADD patches /patches
 ARG GKWILLIE_TOKEN
 RUN apt-get update && \
     apt-get install -y --no-install-suggests --no-install-recommends gcc g++ patch && \
     sed -i "s/git+ssh:\/\/git@/git+https:\/\/gkwillie:$GKWILLIE_TOKEN@/g" server/requirements.txt && \
+    echo "Installing Python packages" && \
     pip3 install --no-cache-dir -r /server/requirements.txt && \
     sed -i "s/git+https:\/\/gkwillie:$GKWILLIE_TOKEN@/git+ssh:\/\/git@/g" server/requirements.txt && \
     pip3 uninstall -y flask && \
     patch /usr/local/lib/python*/dist-packages/prometheus_client/exposition.py /patches/prometheus_client.patch && \
+    chown -R worker:worker /usr/local/lib/python* && \
```

Comment:
Doesn't this reduce security? How about adding the user in the end?

Reply:
Maybe we can also avoid this chown, installed packages should have pyc files, and we are not interested in worker user creating them anyway.
What do you mean by adding the user in the end anyway ?

Reply:
I mean leave everything as-is and execute `adduser` right before `USER` and `ENTRYPOINT`.

btw, we should test `sudo` or `su` with that new user, otherwise, we cannot sacrifice debuggability.

Reply:
We need write permission at least in `/server/` to test migrations I think.
To debug from outside it should be ok to do `docker exec --user root` so enabling sudo is not needed

Reply:
Na-nah, we are under kubernetes in GKE. What works in docker will not work there.

Reply:
The topic is live debugging in production, which I had to do numerous times and which is the most important property we cannot lose.

Reply:
Regarding the write permission in `/server`, we can make it more fine-grained in just those directories that require RW.

Reply:
Right, `kubectl exec` doesn't allow to propagate `-u`
https://github.com/kubernetes/kubectl/blob/a0af655b7abaf06983c99ad5a4dc8f02ae5eb3e5/pkg/cmd/exec/exec.go#L100

Reply:
Maybe we could set a root password ?